### PR TITLE
Introduce the general scheme instead of interoperability

### PR DESCRIPTION
**Pull request by gtklocker**
Depends on #23.

### CODE REVIEW

File: introduction.tex [hunk 1]
```diff
@@ -1,53 +1,22 @@
...
+A user who wishes to burn her coins generates an address which we call a \emph{burn address} that encodes some metadata (called the \emph{tag}). She can then proceed to send any amount of cryptocurrency to the burn address. After burning her cryptocurrency, she may prove to any interested party that she has actually irrevocably destroyed the cryptocurrency in question.
```
Comment (dionyziz) [inline introduction.tex]: Shorten:

* you don't need the () around "called the \emph{tag}"
* remove the "has actually"
* She can then proceed -> She then proceeds
* she may prove -> she proves
* Remove the "in question"

File: introduction.tex [hunk 2]
```diff
@@ -56,21 +25,18 @@ \section{Introduction}\label{section:introduction}
     define it as a protocol which consists of two algorithms, a burn address \emph{generator} and a burn address \emph{verifier}. We put forth the foundational properties which make for secure burn protocols, namely \emph{unspendability}, \emph{binding}, and \emph{uncensorability}.
     \item \textbf{Novel construction.} We propose a novel and simple construction which is flexible and can be adapted for use in existing cryptocurrencies, as long as they use public key hashes for address generation. To our knowledge, all popular cryptocurrencies are
     compatible with our scheme. We prove our construction secure in the Random Oracle model.
-    \item \textbf{Experimental results.} We provide a compehensively tested production-grade implementation of our burn verifier in Ethereum
+    \item \textbf{Bootstrapping mechanism.} We propose a cryptocurrency proof-of-burn bootstrapping mechanism which is both miner-isolated and decentralized by making use of Non-Interactive Proofs of Proof-of-Work (NIPoPoWs)~\cite{nipopows}. Our mechanism in principle allows burning from any proof-of-work-based cryptocurrency.
```
Comment (dionyziz) [inline introduction.tex]: What is miner isolation?

File: introduction.tex [hunk 3]
```diff
@@ -1,53 +1,22 @@
 \section{Introduction}\label{section:introduction}
+Proof-of-burn was invented in 2012 by Iain Stewart~\cite{stewart} as a mechanism for destroying cryptocurrency in an irrevocable and provable manner. It has since been used as a consensus mechanism similar to proof-of-stake (Slimcoin~\cite{slimcoin}), as a mechanism for establishing identity (OpenBazaar~\cite{zindros2016trust}), notarization (Carbon dating~\cite{clark2012commitcoin} and OpenTimestamps~\cite{todd2016opentimestamps}) and bootstrapping a new cryptocurrency (Counterparty~\cite{counterparty}).
```
Comment (dionyziz) [inline introduction.tex:2]: Move this paragraph to related work

File: introduction.tex [hunk 4]
```diff
@@ -1,53 +1,22 @@
...
-
-This chicken-and-egg situation presents the need for a solution that circumvents exchanges and allows users to acquire the cryptocurrency in a usable manner, in order for it to gain traction. We recommend utilizing proof-of-burn in order to can allow users to obtain capital on a new cryptocurrency, automatically and without trusted third parties, by burning a legacy proof-of-work cryptocurrency.
-
-We wish to allow users to exchange a proof-of-work based cryptocurrency for a new cryptocurrency. We call the former the \emph{source cryptocurrency}, and the blockchain this cryptocurrency is based on the \emph{source blockchain}. We call the latter the \emph{target cryptocurrency} and its corresponding blockchain the \emph{target blockchain}. The target blockchain may support burning from multiple source blockchains.
```
Comment (dionyziz) [inline introduction.tex:7]: I think we need these paragraphs in the consumption section in shortened form.

File: introduction.tex [hunk 5]
```diff
@@ -1,53 +1,22 @@
 \section{Introduction}\label{section:introduction}
```
Comment (dionyziz) [inline introduction.tex:1]: Write an introduction about what proof of burn is for the uninitiated.

**Review by dionyziz — CHANGES_REQUESTED** [review]

File: introduction.tex (hunk 4 above)
Comment (gtklocker) [reply introduction.tex:7]: they are moved, not deleted

Comment (gtklocker) [reply introduction.tex:7]: and shortened where appropriate

File: conclusion.tex [hunk 6]
```diff
@@ -1 +1,5 @@
-\section{Conclusion}
\ No newline at end of file
+\section{Conclusion}
+
+For bootstrapping cryptocurrencies, proof-of-burn has been previously used in systems where direct
+observation allowed for the system to give rise to the new tokens such as
+Counterparty~\cite{counterparty}, a velvet fork of Bitcoin. The construction posed by Counterparty only satisfies two of the desired properties, namely unspendability and binding. It does not satisfy usability as the user needs to create a special type of transaction with an $\opreturn$ output. Because of this it also does not satisfy uncensorability, since this form of transaction can be easily detected by miners. Finally, it does not satisfy miner-isolation as users need to be connected to the Bitcoin network in order for the burn transactions to be verified.
```
Comment (dionyziz) [inline conclusion.tex]: Let's drop this section completely

File: consumption.tex [hunk 7]
```diff
@@ -1,10 +1,20 @@
 \section{Consumption}
 
+Over the last 5 years there has been an explosion of innovation through new cryptocurrencies. Unfortunately, it is very hard for a new cryptocurrency to gain traction. Without traction, no market depth ensues, a cryptocurrency will have difficulty getting listed in any exchange. But without being listed in any exchange, a cryptocurrency cannot gain sufficient traction. The cryptocurrency can only be sold over-the-counter (OTC) by its miners or stakeholders, but that market is very limited.
```
Comment (dionyziz) [inline consumption.tex]: _and_ a cryptocurrency will have difficulty

Comment (dionyziz) [reply consumption.tex]: Shorten:

* will have difficulty -> has difficulty

Comment (dionyziz) [reply consumption.tex]: I think we can drop the sentence about OTC

File: consumption.tex [hunk 8]
```diff
@@ -1,10 +1,20 @@
...
+
+This chicken-and-egg situation presents the need for a solution that circumvents exchanges and allows users to acquire the cryptocurrency in a usable manner, in order for it to gain traction. We recommend utilizing proof-of-burn in order to allow users to obtain capital on a new cryptocurrency, automatically and without trusted third parties, by burning a legacy cryptocurrency.
```
Comment (dionyziz) [inline consumption.tex]: recommend -> propose

We don't recommend here, we make a proposal that each cryptocurrency can assess on their own to see if it's suitable for them.

File: consumption.tex [hunk 9]
```diff
@@ -1,10 +1,20 @@
...
+
+We wish to allow users to exchange a legacy cryptocurrency for a new cryptocurrency. We call the former the \emph{source cryptocurrency}, and the blockchain this cryptocurrency is based on the \emph{source blockchain}. We call the latter the \emph{target cryptocurrency} and its corresponding blockchain the \emph{target blockchain}. The target blockchain may support burning from multiple source blockchains.
```
Comment (dionyziz) [inline consumption.tex]: shorten:

* and the blockchain this cryptocurrency is based on  -> and its blockchain

File: consumption.tex [hunk 10]
```diff
@@ -1,10 +1,20 @@
...
 
-A user who wishes to acquire a target cryptocurrency first has to form a burn address which encodes her receiving address on the target blockchain by using it as a tag. The user then sends an amount of the source cryptocurrency to that address. She then proceeds to submit a proof of this burn to a smart contract on the target blockchain, where the burn is verified and the user is credited an equivalent amount of target cryptocurrency on her receiving address.
+\noindent
+\textbf{Workflow.}
+A user who wishes to acquire a target cryptocurrency first has to form a burn address which encodes her receiving address on the target blockchain by using it as a tag. She then sends an amount of the source cryptocurrency to that address. She submits a proof of this burn to a smart contract on the target blockchain, where it is verified and she is credited an equivalent amount of target cryptocurrency on her receiving address. Proof of burn verification happens in either a centralized manner which is lighter on computation but requires a federation, or in a decentralized manner using Non-Interactive Proofs of Proof-of-Work (NIPoPoWs).
```
Comment (dionyziz) [inline consumption.tex]: shorten:

* has to form -> forms

File: introduction.tex [hunk 11]
```diff
@@ -1,53 +1,22 @@
...
-
+We define the following properties for a proof-of-burn protocol:
 \begin{itemize}
-    \item \textbf{Unspendability.} No one should be able to spend the amount burned on the source blockchain.
-    \item \textbf{Binding.} The burn must commit to details of the receiver on the target blockchain.
-    \item \textbf{Uncensorability.} Miners who do not agree with this scheme on the source blockchain should not be able to censor burn transactions.
-    \item \textbf{Usability.} The user should be able to create a burn transaction using her regular source cryptocurrency wallet.
-    \item \textbf{Miner-isolation.} No miner on the target blockchain should be required to connect to the network of any of the source blockchains. % TODO mv
+    \item \textbf{Unspendability.} No one should be able to spend the amount burned.
```
Comment (dionyziz) [inline introduction.tex]: should be able to -> can

File: introduction.tex [hunk 12]
```diff
@@ -1,53 +1,22 @@
...
+    \item \textbf{Binding.} The burn must commit only to a single tag.
```
Comment (dionyziz) [inline introduction.tex]: must commit -> commits

File: introduction.tex [hunk 13]
```diff
@@ -1,53 +1,22 @@
...
+    \item \textbf{Uncensorability.} Miners who do not agree with this scheme should not be able to censor burn transactions.
```
Comment (dionyziz) [inline introduction.tex]: should not be able to -> cannot

File: introduction.tex [hunk 14]
```diff
@@ -1,53 +1,22 @@
...
+    \item \textbf{Usability.} The user should be able to create a burn transaction using her regular cryptocurrency wallet.
```
Comment (dionyziz) [inline introduction.tex]: should be able to -> can

**Review by dionyziz — CHANGES_REQUESTED** [review]

**Comment by gtklocker** [comment]
ready

File: consumption.tex [hunk 15]
```diff
@@ -1,10 +1,20 @@
 \section{Consumption}
 
+Over the last 5 years there has been an explosion of innovation through new cryptocurrencies. Unfortunately, it is very hard for a new cryptocurrency to gain traction. Without traction, no market depth ensues and a cryptocurrency has difficulty getting listed in any exchange. But without being listed in any exchange, a cryptocurrency cannot gain sufficient traction.
+
+This chicken-and-egg situation presents the need for a solution that circumvents exchanges and allows users to acquire the cryptocurrency in a usable manner, in order for it to gain traction. We propose utilizing proof-of-burn in order to allow users to obtain capital on a new cryptocurrency, automatically and without trusted third parties, by burning a legacy cryptocurrency.
```
Comment (dionyziz) [inline consumption.tex]: remove "in order"

File: consumption.tex [hunk 16]
```diff
@@ -1,10 +1,20 @@
...
 
-A user who wishes to acquire a target cryptocurrency first has to form a burn address which encodes her receiving address on the target blockchain by using it as a tag. The user then sends an amount of the source cryptocurrency to that address. She then proceeds to submit a proof of this burn to a smart contract on the target blockchain, where the burn is verified and the user is credited an equivalent amount of target cryptocurrency on her receiving address.
+\noindent
+\textbf{Workflow.}
+A user who wishes to acquire a target cryptocurrency first forms a burn address which encodes her receiving address on the target blockchain by using it as a tag. She then sends an amount of the source cryptocurrency to that address. She submits a proof of this burn to a smart contract on the target blockchain, where it is verified and she is credited an equivalent amount of target cryptocurrency on her receiving address. Proof-of-burn verification happens in either a centralized manner which is lighter on computation but requires a federation, or in a decentralized manner using Non-Interactive Proofs of Proof-of-Work (NIPoPoWs).
```
Comment (dionyziz) [inline consumption.tex]: ...forms a burn address _valid in the source blockchain_ which...

File: consumption.tex [hunk 17]
```diff
@@ -1,10 +1,20 @@
...
+
+Target blockchain miners need not be connected to every other source blockchain network. We call this property \emph{miner-isolation} and we propose methods to achieve it.
 
-In accordance to the terminology laid out in~\cite{pow-sidechains}, we call this party the \emph{prover} and we call the smart contract the \emph{verifier}.
+In accordance to the terminology laid out in~\cite{pow-sidechains}, we call the user the \emph{prover} and we call the smart contract the \emph{verifier}.
```
Comment (dionyziz) [inline consumption.tex]: remove second "we call"

**Review by dionyziz — APPROVED** [review]

**Pull request merged**